src/sqlite3/result_set.cr

```diff
@@ -44,25 +44,57 @@ class SQLite3::ResultSet < DB::ResultSet
         raise Exception.new(sqlite3_statement.sqlite3_connection)
       end
     @column_index += 1
-    value
+    value.as(Float64 | Int64 | Slice(UInt8) | String | Time | Nil)
```

Comment:
I narrowed it down a little bit more. Why is `.as(DB::ResultSet)` called in the changes of `driver_spec.cr`? 

Ref: https://github.com/crystal-lang/crystal/issues/4247